hash-parser/src/error.rs

```diff
@@ -0,0 +1,69 @@
+//! Compiler error reporting
+//
+// All rights reserved 2021 (c) The Hash Language authors
+
+use crate::{grammar::Rule, location::Location};
+use std::fmt;
+
+/// Error message prefix
+const ERR: &str = "\x1b[31m\x1b[1merror\x1b[0m";
+
+/// Hash ParseError enum represnting the variants of possible errors.
+#[derive(Debug, Clone)]
+pub enum ParseError {
+    IoError {
+        filename: String,
+    },
+    Parsing {
+        positives: Vec<Rule>,
+        negatives: Vec<Rule>,
+        location: Location,
+    },
+    AstGeneration {
+        rule: Rule,
+        location: Location,
+    },
+}
+
+/// Convert a [pest::error::Error] into a [ParseError]
+impl<'a> From<pest::error::Error<Rule>> for ParseError {
```

Comment:
Why the lifetime parameter here? Does `cargo clippy` not say anything about it?

Reply:
Clippy reports a few errors before it reaches this module, so it's unkown.